wallets/wrap.js

```diff
@@ -0,0 +1,169 @@
+import { createHodlInvoice, getHeight, parsePaymentRequest } from 'ln-service'
+import { estimateRouteFee } from '../api/lnd'
+
+const MIN_OUTGOING_MSATS = BigInt(900) // the minimum msats we'll allow for the outgoing invoice
+const MAX_OUTGOING_MSATS = BigInt(9_000_000_000) // the maximum msats we'll allow for the outgoing  invoice
+const MAX_EXPIRATION_INCOMING_MSECS = 900_000 // the maximum expiration time we'll allow for the incoming invoice
+const INCOMING_EXPIRATION_BUFFER_MSECS = 300_000 // the buffer enforce for the incoming invoice expiration
+const MIN_INCOMING_CLTV_DELTA = 200 // the minimum cltv delta we'll allow for the incoming invoice
+const MAX_OUTGOING_CLTV_DELTA = 360 // the maximum cltv delta we'll allow for the outgoing invoice
+export const MIN_SETTLEMENT_CLTV_DELTA = 80 // the minimum blocks we'll leave for settling the incoming invoice
+const FEE_ESTIMATE_TIMEOUT_SECS = 5 // the timeout for the fee estimate request
+const MAX_FEE_ESTIMATE_PERCENT = 0.02 // the maximum fee relative to outgoing we'll allow for the fee estimate
+const OUTGOING_ROUTING_ESTIMATE_MULT = 1.1 // pad the budget for outgoing routing
+const ZAP_SYBIL_FEE_MULT = 10 / 9 // the fee for the zap sybil service
+/*
+  The wrapInvoice function is used to wrap an outgoing invoice with the necessary parameters for an incoming hold invoice.
+
+  @param bolt11 {string} the bolt11 invoice to wrap
+  @param description {string} the description to use for the incoming invoice
+  @param descriptionHash {string} the description hash to use for the incoming invoice
+  @returns {
+    invoiceParams: {
+      id: string,
+      description: string,
+      description_hash: string,
+      expires_at: Date,
+      cltv_delta: number,
+      mtokens: string
+    },
+    outgoingMaxFeeMsat: number
+  }
+*/
```

Comment:
This comment is out of date. We now return a hold invoice rather than `invoiceParams`.